AnimationSystem/AnimationSystem.cc

```diff
@@ -0,0 +1,1074 @@
+          else if (boneName == "Right_littleFinger3") boneIndexes.Right_littleFinger3 = i;
+          else if (boneName == "Right_leg") boneIndexes.Right_leg = i;
+          else if (boneName == "Right_knee") boneIndexes.Right_knee = i;
+          else if (boneName == "Right_ankle") boneIndexes.Right_ankle = i;
+          else if (boneName == "Right_toe") boneIndexes.Right_toe = i;
+          else if (boneName == "Left_leg") boneIndexes.Left_leg = i;
+          else if (boneName == "Left_knee") boneIndexes.Left_knee = i;
+          else if (boneName == "Left_ankle") boneIndexes.Left_ankle = i;
+          else if (boneName == "Left_toe") boneIndexes.Left_toe = i;
+        }
+      }
+
+      // -------------------------------------------------------------------------
+
+      jsonStr += "[";
+      for (unsigned int i = 0; i < declarations.size(); i++) {
+        AnimationGroupDeclaration declaration = declarations[i];
+        std::vector<Animation *> animationGroup;
+
+        jsonStr += "{";
+        jsonStr += "\"name\":";
+        jsonStr += "\"" + declaration.groupName + "\"";
+        jsonStr += ",";
+        jsonStr += "\"index\":";
+        jsonStr += std::to_string(declaration.index);
+        jsonStr += ",";
+        jsonStr += "\"animations\":";
+        jsonStr += "[";
+        for (unsigned int j = 0; j < declaration.animationDeclarations.size(); j++) {
+          AnimationDeclaration animationDeclaration = declaration.animationDeclarations[j];
+          animationGroup.push_back(animationAll[animationDeclaration.fileName]);
+
+          jsonStr += "{";
+          jsonStr += "\"keyName\":";
+          jsonStr += "\"" + animationDeclaration.keyName + "\"";
+          jsonStr += ",";
+          jsonStr += "\"index\":";
+          jsonStr += std::to_string(animationDeclaration.index);
+          jsonStr += ",";
+          jsonStr += "\"fileName\":";
+          jsonStr += "\"" + animationDeclaration.fileName + "\"";
+          jsonStr += "}";
+          if (j != declaration.animationDeclarations.size() - 1) jsonStr += ",";
+        }
+        animationGroups.push_back(animationGroup);
+        jsonStr += "]";
+        jsonStr += "}";
+        if (i != declarations.size() - 1) jsonStr += ",";
+      }
+      jsonStr += "]";
+
+      // -------------------------------------------------------------------------
+
+      defaultSitAnimationIndex = sitAnimationIndexes.Chair;
+      defaultEmoteAnimationIndex = emoteAnimationIndexes.Angry;
+      defaultDanceAnimationIndex = danceAnimationIndexes.Dansu;
+      defaultHoldAnimationIndex = holdAnimationIndexes.Pick_up_idle;
+      defaultActivateAnimationIndex = activateAnimationIndexes.Grab_forward;
+      defaultNarutoRunAnimationIndex = narutoRunAnimationIndexes.NarutoRun;
+
+      // -------------------------------------------------------------------------
+
+      activateSpeedFactors.push_back(1.2); // Grab_forward
```

Comment:
Since this is highly dependent on the animations, this code should not exist in the C++.

If the animations are the wrong speed then the animation should be updated.

Reply:
Updated animation `.fbx` files directly, instead of using `activateSpeedFactors`.
https://github.com/upstreet-labs/app/pull/116/commits/13b7f9c828e7c43a35037b6193733f9bc47c194b
https://github.com/upstreet-labs/app/pull/116/commits/b8a20a6d5fe47ac1e6f0f84fa9a07c5245f89152
https://github.com/webaverse/app-wasm/pull/75/commits/f99e1f672b9390aa5940634178286e17fa79f52b